clash-prelude/src/Clash/Explicit/Reset.hs

```diff
@@ -52,6 +55,53 @@ import           GHC.TypeLits (type (+))
 >>> import Clash.Explicit.Prelude
 -}
 
+-- | A deasserted reset
+noReset :: KnownDomain dom => Reset dom
+noReset = unsafeFromHighPolarity (pure False)
+
+-- | Output reset will be asserted when either one of the input resets is
+-- asserted
+orReset ::
+  forall dom .
+  ( KnownDomain dom
+  , DomainResetKind dom ~ 'Synchronous ) =>
+  Reset dom ->
+  Reset dom ->
+  Reset dom
+orReset = unsafeOrReset
+
+-- | Output reset will be asserted when either one of the input resets is
+-- asserted. This function is considered unsafe because it can be used on
+-- domains with components whose resets are level sensitive, while use of this
+-- function can introduce glitches.
```

Comment:
```suggestion
-- domains with components with asynchronous resets, where use of this function
-- can introduce glitches triggering a reset.
```

I like consistency in terminology; we mainly call it (a)synchronous resets. Let's not require of our users that they know all the synonyms... also, let's be a bit more verbose about those glitches.